independent-projects/qute/core/src/main/java/io/quarkus/qute/Expressions.java

```diff
@@ -0,0 +1,107 @@
+package io.quarkus.qute;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+
+public final class Expressions {
+
+    public static final String TYPECHECK_NAMESPACE_PLACEHOLDER = "$$namespace$$";
+
+    static final String LEFT_BRACKET = "(";
+    static final String RIGHT_BRACKET = ")";
+
+    private Expressions() {
+    }
+
+    public static boolean isVirtualMethod(String value) {
+        return value.indexOf(LEFT_BRACKET) != -1;
+    }
+
+    public static String parserVirtualMethodName(String value) {
+        int start = value.indexOf(LEFT_BRACKET);
+        return value.substring(0, start);
+    }
+
+    public static List<String> parseVirtualMethodParams(String value) {
+        int start = value.indexOf(LEFT_BRACKET);
+        if (start != -1 && value.endsWith(RIGHT_BRACKET)) {
+            String params = value.substring(start + 1, value.length() - 1);
+            return splitParts(params, Expressions::isParamSeparator);
+        }
+        throw new IllegalArgumentException("Not a virtual method: " + value);
```

Comment:
Any reason why you have this check here and not in `parserVirtualMethodName`?

Reply:
Sorry, I don't remember. Anyway, this code will need some more consolidation :-).